### PR TITLE
test(core): add router tests and fix edge cases

### PR DESCRIPTION
**Pull request by ericmarcos**

### CODE REVIEW

File: packages/botonic-core/src/router.js [hunk 1]
```diff
@@ -6,7 +6,34 @@ export class Router {
     this.routes = routes
   }
 
-  processInput(input, session, lastRoutePath) {
+  /*processInput(input, session, lastRoutePath = null, redirects = 0) {
```
Comment (dpinol) [inline packages/botonic-core/src/router.js]: remove?

File: packages/botonic-core/src/router.test.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+const textInput = { type: 'text', data: 'hi' }
+const textInputComplex = { type: 'text', data: 'Cömplêx input &% 🚀' }
+const textPayloadInput = { type: 'text', data: 'hi', payload: 'foo' }
+const postbackInput = { type: 'postback', payload: 'foo' }
+
+describe('Bad router initialization', () => {
+  test('empty routes throw TypeError', () => {
+    let router = new Router([])
+    expect(() => router.processInput2(textInput)).toThrow(TypeError)
```
Comment (dpinol) [inline packages/botonic-core/src/router.test.js]: processInput2-> processInput
